framework/include/utils/InputParameters.h

```diff
@@ -968,6 +968,7 @@ class InputParameters : public Parameters
   // These are the only objects allowed to _create_ InputParameters
   friend InputParameters emptyInputParameters();
   friend class InputParameterWarehouse;
+  friend class Parser;
```

Comment:
@permcody I couldn't make it work to only have `Parser::setVectorParameter<PostprocessorName, PostprocessorName>(...)` a friend. Not exactly sure why it didn't work, but it may be that `Parser.h` includes `InputParameters.h` and I now have to include `Parser.h` in `InputParameters.h` as well.

Reply:
Yeah - that would do it. There's no work around to that issue.